libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -406,22 +407,24 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         )
 
         ifNonNull(accessToken) { token ->
-            logger.d(
-                MapboxNavigationTelemetry.TAG,
-                Message("MapboxMetricsReporter.init from MapboxNavigation main")
-            )
-            MapboxMetricsReporter.init(
-                navigationOptions.applicationContext,
-                token,
-                obtainUserAgent()
-            )
-            MapboxMetricsReporter.toggleLogging(navigationOptions.isDebugLoggingEnabled)
-            MapboxNavigationTelemetry.initialize(
-                this,
-                navigationOptions,
-                MapboxMetricsReporter,
-                logger,
-            )
+            runInTelemetryContext { telemetry ->
+                logger.d(
+                    telemetry.TAG,
+                    Message("MapboxMetricsReporter.init from MapboxNavigation main")
+                )
+                MapboxMetricsReporter.init(
+                    navigationOptions.applicationContext,
+                    token,
+                    obtainUserAgent()
+                )
```

Comment:
The problem here is that if the telemetry is disabled initially, we won't initialize the metrics reporter, especially the `lateinit var` in `MapboxMetricsReporter#mapboxTelemetry`.

So if later during the session telemetry becomes enabled, _other_calls would go through and crash on the `MapboxMetricsReporter#mapboxTelemetry` not being initialized, I believe `MapboxNavigation#onDestroy` could cause that but there could also be other places.

@RingerJK could you double check that before this contribution is released?

Reply:
`TelemetryEnabler#isEventsEnabled` checks if key `com.mapbox.EnableEvents` key defined, it's not changing in runtime.

Reply:
Oh, I see, this doesn't cover the case if the telemetry is disabled in the runtime then (for example with the maps attribution button, and saved to shared prefs). This means that the merged PR should be safe to ship :+1: 

Could you cut a separate ticket for runtime handler though?